pkg/billing/client.go

```diff
@@ -40,24 +37,9 @@ func (b *Batch) Count() int {
 	return len(b.events)
 }
 
-func (b *Batch) idempotenize(key string) string {
-	if key != "" {
-		return key
-	}
-
-	return fmt.Sprintf("Host<%s>:ID<%s>:T<%s>", b.c.hostname, uuid.NewString(), time.Now().Format(time.RFC3339))
-}
-
-func (b *Batch) AddAbsoluteEvent(e AbsoluteEvent) {
-	e.Type = "absolute"
-	e.IdempotencyKey = b.idempotenize(e.IdempotencyKey)
-	b.events = append(b.events, &e)
-}
-
-func (b *Batch) AddIncrementalEvent(e IncrementalEvent) {
-	e.Type = "incremental"
-	e.IdempotencyKey = b.idempotenize(e.IdempotencyKey)
-	b.events = append(b.events, &e)
+func (b *Batch) Add(e enrichable) {
+	Enrich(b, e) // Realistically, we're already calling Enrich before Add, but this is a good safety mechanism.
```

Comment:
a better safety mechanism might be a test ;)

I think I'd opt for calling it in only one of these places (my vote is this one, with some minor modifications)

Reply:
It can't be just this one because we're trying to log `IdempotencyKey` in pkg/agent/billing

Reply:
put the logging in the add/batch method?

Reply:
That feels weird, a little bit "spooky-action-at-a-distance"-y. Basically, I don't think the billing client should log anything, and that should be left to the API user.

(next PR will also make `Batch.events` public, in order to e.g. decrease size if we get a 413 / context expired / etc.)